messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQResourceAdapter.java

```diff
@@ -79,4 +85,72 @@ protected BroadcastEndpointFactory createBroadcastEndpointFactory(ConnectionFact
     private static ServiceContainer currentServiceContainer() {
         return (System.getSecurityManager() == null) ? CurrentServiceContainer.getServiceContainer() : AccessController.doPrivileged(CurrentServiceContainer.GET_ACTION);
     }
+
+    @Override
+    public WorkManager getWorkManager() {
+        return new NamingWorkManager(super.getWorkManager());
+    }
+
+    private class NamingWorkManager implements WorkManager {
```

Comment:
We should add a method to wrap the Work instance, and use it on all public methods, not just the specific scheduleWork, just in case changes are done at Artemis side, regarding how it uses the WorkManager... E.g.

    private Work wrapWork(Work work) {
      return new WorkWrapper(NamespaceContextSelector.getCurrentSelector(), work);
    }

    @Override
    public void scheduleWork(Work work) throws WorkException {
       delegate.scheduleWork(wrapWork(work));
    }